src/components/Extensions/models.ts

```diff
@@ -136,6 +136,12 @@ export interface ExtensionSpec {
   externalServices?: ExternalService[];
 }
 
+export interface Extension extends ExtensionSpec {
```

Comment:
One more time for the record... I'm not super excited about extending here, I'd prefer composition - but don't consider this a blocker.

Reply:
I overall support, but want to do it a bit later, once I have better understanding of things